Documentation/Plugins.md

```diff
@@ -203,13 +207,9 @@ struct MyBuildToolPlugin: BuildToolPlugin {
 }
 ```
 
-In the case of prebuild commands, any dependencies must be binary targets, since these commands run before the build starts.
-
-Note that a build tool plugin can return a combination of build tool commands and prebuild commands.  After the plugin runs, any build commands are incorporated into the build graph, which may result in changes that require commands to run during the subsequent build.
-
-Any prebuild commands are run after the plugin runs but before the build starts, and any files that are in the prebuild command's declared `outputFilesDirectory` will be evaluated as if they had been source files in the target.  The prebuild command should add or remove files in this directory to reflect the results of having run the command.
+A prebuild command has no inputs, so it will never be re-run due to changes in source files. The only trigger for re-running a prebuild command is a change to a declared dependendencies, which can only be (prebuilt) binary targets, since these commands run before any other targets have been built.
 
-The current version of the Swift Package Manager supports generated Swift source files and resources as outputs, but it does not yet support non-Swift source files.  Any generated resources are processed as if they had been declared in the manifest with the `.process()` rule.  The intent is to eventually support any type of file that could have been included as a source file in the target, and to let the plugin provide greater controls over the downstream processing of generated files.
+Any `.swift` files that are outputs of build commands or prebuild commands will be compiled into the target using the plugin. Currently, any other output files are treated as resources and processed as if they had been declared in the manifest with the `.process()` rule.  The intent is to eventually support any type of file that could have been included as a source file in the target, and to let the plugin provide greater controls over the downstream processing of generated files.
```

Comment:
```suggestion
The current version of the Swift Package Manager supports generated Swift source files and resources as outputs, but it does not yet support non-Swift source files. Any `.swift` files that are outputs of build commands or prebuild commands will be compiled into the target using the plugin. Currently, any other output files (including non-Swift source files) are treated as resources and processed as if they had been declared in the manifest with the `.process()` rule.  The intent is to eventually support any type of file that could have been included as a source file in the target, and to let the plugin provide greater controls over the downstream processing of generated files.
```

Reply:
Using the phrase “non-Swift source files” or “Swift source files” concerns me, because the actual issue is not what programming language is used in the file, but the file's extension.  What I wrote was absolutely unambiguous, but if you put C source in a `.swift` file, it will not in fact be treated as a resource, and if you put Swift source in a `.foo` file, it will, and that conflicts with the language you've added.

Reply:
How about 

The current version of the Swift Package Manager supports generated .swift files (which will be treated as Swift source) and resources as outputs, but it does not yet support other types of source files.

Reply:
It's OK by itself, but then when you put it with the next sentence it's redundant and verbose.  It seems like the only thing you're trying to add is an _explicit_ mention that compiling output files in languages other than Swift isn't supported.  Does this accomplish what you want?
 
```suggestion
Any `.swift` files that are outputs of build commands or prebuild commands will be treated as Swift source files and compiled into the target being build by the plugin. Currently, compilation of output files in other source langauges isn't supported, so any other output files are treated as resources and processed as if they had been declared in the manifest with the `.process()` rule.  The intent is to eventually support any type of file that could have been included as a source file in the target, and to let the plugin provide greater controls over the downstream processing of generated files.
```

BTW, I don't understand what “and to let the plugin provide greater controls over the downstream processing of generated files” might actually mean, so if I were you I would consider striking or clarifying it.  The only reason I didn't do that is because in this context, where you're just talking about a future SPM that doesn't exist yet, I can easily know the information isn't relevant to me.  I would also suggest you consider moving the whole sentence into a footnote, because it really is irrelevant to almost anyone who is trying to understand what a build tool is and what it can do.

Reply:
> Does this accomplish what you want?

yes